### ISSUE TITLE
Unable to generate passwords with >72 characters

### ISSUE THREAD

**Issue by henryde** [hashicorp/terraform-provider-random#396]
### Terraform CLI and Provider Versions

terraform: 1.4.2
provider: 3.5.0

### Terraform Configuration

```terraform
resource "random_password" "test" {
  length  = 73
}
```

### Expected Behavior

generating longer passwords was possible with 3.4.3

### Actual Behavior

```
╷
│ Error: Hash Generation Error
│
│   with random_password.test,
│   on test.tf line 1, in resource "random_password" "test":
│    1: resource "random_password" "test" {
│
│ While attempting to generate a hash from the password an error occurred.
│
│ Verify that the state contains a populated 'result' field, using 'terraform state show', and retry the operation
│
│ Original Error: bcrypt: password length exceeds 72 bytes
╵
```

### Steps to Reproduce

1. `terraform apply`

### How much impact is this issue causing?

Medium

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ashhammond** [comment]
We are also seeing the exact same issue with the latest version of the provider. Rolling back and testing as suggested above.

**Comment by ashhammond** [comment]
As a workaround we have now pinned the random provider to 3.4.3 and functionality is restored.

**Comment by austinvalle** [comment]
Hi there @henryde 👋🏻 , thanks for submitting the issue and sorry you ran into this bug.

We've merged a fix for this bug with #397 and released `v3.5.1` of the random provider with this bug fix. You should see that provider version become available in the next hour or so.

**Comment by ashhammond** [comment]
thanks @austinvalle for the quick fix.
